src/v0/destinations/marketo/util.js

```diff
@@ -122,17 +123,27 @@ const marketoApplicationErrorHandler = (marketoResponse, sourceMessage, destinat
 const nestedResponseHandler = (marketoResponse, sourceMessage) => {
   const checkStatus = (res) => {
     const { status } = res;
-    if (status && status !== 'updated' && status !== 'created' && status !== 'added') {
+    const allowedStatus = ['updated', 'added', 'removed', 'created'];
+    if (
+      status &&
+      !allowedStatus.includes(status)
+      // we need to check the case where the id are not in list
+    ) {
       const { reasons } = res;
       let statusCode = 400;
-      if (reasons && RECORD_LEVEL_ABORTBALE_ERRORS.includes(reasons[0].code)) {
-        statusCode = 400;
-      } else if (reasons && MARKETO_ABORTABLE_CODES.includes(reasons[0].code)) {
-        statusCode = 400;
-      } else if (reasons && MARKETO_THROTTLED_CODES.includes(reasons[0].code)) {
-        statusCode = 429;
-      } else if (reasons && MARKETO_RETRYABLE_CODES.includes(reasons[0].code)) {
-        statusCode = 500;
+      if (reasons) {
+        const errorCodesFromDest = reasons.map((reason) => reason.code);
+        const filteredErrorCode = errorCodesFromDest.find(
+          (errorCode) => !ERROR_CODE_TO_PASS.includes(errorCode),
+        );
+        if (!filteredErrorCode) {
+          return;
+        }
+        if (MARKETO_THROTTLED_CODES.includes(filteredErrorCode.code)) {
+          statusCode = 429;
+        } else if (MARKETO_RETRYABLE_CODES.includes(filteredErrorCode.code)) {
+          statusCode = 500;
+        }
```

Comment:
What about abortable codes?

Reply:
We are/were taking the error as abortable by default